### PR TITLE
[4.0] Namespace System - Privacy Consent plugin

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

Namespaces System - Privacy Consent plugin.

### Testing Instructions

Apply patch. Delete `libraries/autoload_psr4.php` file.
Enable System - Confirm Consent plugin.
Login to frontend.
When redirected to profile form to accept consent check that Privacy Policy field appears correctly.

### Documentation Changes Required

IDK.

### CODE REVIEW

**Comment by jwaisner** [comment]
I have tested this item :red_circle: unsuccessfully on ec00dace048fa74856d2a7ceaf42d8e718adeb8c<br /><br />Privacy consent at bottom of profile is missing after applying patch and removing libraries/autoload_psr4.php.

I am noticing that when reversing the patch I am getting a warning that plugins/system/privacyconsent/src/Field/PrivacyField.php does not exists. I am not sure if the patch is not applying correctly or something is missing from the patch.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27645">issues.joomla.org/tracker/joomla-cms/27645</a>.</sub>

**Comment by SharkyKZ** [comment]
Sounds like the patch wasn't applied correctly. Do you have GitHub Authentication set up in Patchtester? If not, maybe running into request limits.

**Comment by jwaisner** [comment]
Even using and IDE (Eclipse) to apply from the PR it still has a problem reverting the patch.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27645">issues.joomla.org/tracker/joomla-cms/27645</a>.</sub>

File: plugins/system/privacyconsent/privacyconsent.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
 	<namespace path="src">Joomla\Plugin\System\PrivacyConsent</namespace>
 	<files>
 		<filename plugin="privacyconsent">privacyconsent.php</filename>
-		<folder>privacyconsent</folder>
+		<folder>forms</folder>
 		<folder>src</folder>
```
Comment (chmst) [inline plugins/system/privacyconsent/privacyconsent.xml:16]: Shouldn't this be <folder>field</folder>?

Comment (SharkyKZ) [reply plugins/system/privacyconsent/privacyconsent.xml:16]: No, contents of `field` were moved to `src/Field`.

**Comment by jwaisner** [comment]
@SharkyKZ Patch is still referencing a non-existant directory. If we need to move data due to a structure change can you please include those instructions in the patch notes? I tried with Patchtester and via my IDE and both did not work properly.

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 6974caeaff33b00e34fa9afd23e0bb0caa335b45<br /><br />I tested this successfully had to do applied the patch manually in the IDE. 

The structure now is

forms/privacyconsent.php
src/Field/privacy.php
privacyconsent.php
privacyconsent.xml
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27645">issues.joomla.org/tracker/joomla-cms/27645</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 6974caeaff33b00e34fa9afd23e0bb0caa335b45<br /><br />Test is good. Thanks @chmst for the additional instruction.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27645">issues.joomla.org/tracker/joomla-cms/27645</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27645">issues.joomla.org/tracker/joomla-cms/27645</a>.</sub>

**Comment by zero-24** [comment]
Thanks merged. First merge with the new GitHub App, works awesome :-D

**Pull request merged**